inspirehep/utils/references.py

```diff
@@ -42,24 +45,35 @@ def references(self):
             refs_to_get_from_es = [
                 ref['recid'] for ref in references if ref.get('recid')
             ]
-            records_from_es = []
-
-            if refs_to_get_from_es:
-                records_from_es = es.mget(body={
-                    "ids": refs_to_get_from_es
-                }, index="records-hep")['docs']
+            query = IQ(' OR '.join('recid:' + str(ref)
+                                   for ref in refs_to_get_from_es))
+            records_from_es = current_search_client.search(
+                index='records-hep',
+                doc_type='hep',
+                body={"query": query.to_dict()},
+                size=9999,
+                _source=[
+                    'control_number',
+                    'citation_count',
+                    'titles',
+                    'earliest_date',
+                    'authors',
+                    'collaboration',
+                    'corporate_author',
+                    'publication_info'
+                ]
+            )['hits']['hits']
 
             refs_from_es = {
-                ref['_id']: ref['_source'] for ref in records_from_es if '_source' in ref
+                ref['_source']['control_number']: ref['_source'] for ref in records_from_es
             }
             for reference in references:
                 row = []
                 recid = reference.get('recid')
                 ref_record = refs_from_es.get(str(recid)) if recid else None
```

Comment:
Is this always a string in the mapping? I think it would be more future-proof to to `str(ref['_source']['control_number'])` in the dict comprehension.